tamr_client/backup.py

```diff
@@ -0,0 +1,120 @@
+from typing import List
+
+from tamr_client import Backup, response
+from tamr_client._types import Instance, JsonDict, Session, URL
+from tamr_client.exception import TamrClientException
+
+
+class InvalidOperation(TamrClientException):
+    """Raised when attempting an invalid operation.
+    """
+
+    pass
+
+
+class NotFound(TamrClientException):
+    """Raised when referencing a backup that does not exist on the server.
+    """
+
+    pass
+
+
+def _from_json(data: JsonDict) -> Backup:
+    """Make backup from JSON data (deserialize).
+
+    Args:
+        data: Backup JSON data from Tamr server
+    """
+    return Backup(
+        url=data["id"],
+        resource_id=data["relativeId"],
+        path=data["backupPath"],
+        state=data["state"],
+        error_message=data["errorMessage"],
+    )
+
+
+def get_all(session: Session, instance: Instance) -> List[Backup]:
+    """Get information on Tamr backups.
+
+    Args:
+        session: Tamr session
+        instance: Tamr instance
+
+    Returns:
+        A list of Tamr backups
+
+    Raises:
+        backup.NotFound: If no backup found at the specified URL
+    """
+    url = URL(instance=instance, path="backups")
+    r = session.get(str(url))
+    if r.status_code == 404:
+        raise NotFound(str(url))
+    backups = list(_from_json(i) for i in response.successful(r).json())
```

Comment:
nit: if already using a `for`-loop expression, list comprehension is more idiomatic than wrapping in `list(...)`:

```suggestion
    backups = [_from_json(i) for i in response.successful(r).json()]
```

Reply:
changed